source/features/pr-easy-toggle-files.tsx

```diff
@@ -4,23 +4,18 @@ import * as pageDetect from 'github-url-detection';
 
 import features from '.';
 
-const selectorHeaderBar = '.file-header.file-header--expandable';
-
-function toggleFile(event: delegate.Event<Event, HTMLElement>): void {
+function toggleFile(event: Event): void {
 	const elementClicked = event.target as HTMLElement;
-	const headerBar = elementClicked.closest(selectorHeaderBar);
+	const headerBar = elementClicked.closest<HTMLElement>('.file-header.file-header--expandable')!;
 
 	// The clicked element is either the bar itself or one of its 2 children
-	if (
-		elementClicked.isSameNode(headerBar) ||
-		elementClicked.parentElement!.isSameNode(headerBar)
-	) {
-		select('.file-info > button', headerBar as HTMLElement)!.click();
+	if (elementClicked === headerBar || elementClicked.parentElement === headerBar) {
+		select('.file-info > button', headerBar)!.click();
 	}
 }
 
 async function init(): Promise<void> {
-	delegate('#files', selectorHeaderBar, 'click', toggleFile);
+	document.body.addEventListener('click', toggleFile);
```

Comment:
I used that before, but by seeing at other files I thought you would prefer the use of `delegate()`. Care to explain the reason?

Reply:
Thanks for checking the other features too

`delegate` uses `.closest` internally to filter the events, but since we're our listener does it explicitly, we're doing the same thing twice.

Alternatively, if we were to use `delegate`, we'd just need:

```diff
	const headerBar = elementClicked.closest<HTMLElement>('.file-header.file-header--expandable')!;
	const headerBar = element.delegateTarget;
```

They're pretty much equivalent, except that `delegate` does a little more under the hood and therefore since it's easier to avoid it in this case, I removed it.

Reply:
I see, thanks.